### DISCUSSION TITLE
Revision 5 board unflashable?

### DISCUSSION THREAD

**Discussion by ant-ds** [emporia-vue-local/esphome#31] [Q&A] [answered]
Hi,

I am unable to flash the Emporia Vue 2. The board is labeled as revision 5. I am able to flash other ESP32 and 8266 chips, but with this one I get either:
`Failed to connect to Espressif device: No serial data received.`
or `Failed to connect to Espressif device: Invalid head of packet (0xF0): Possible serial noise or corruption.`

What should my next steps be to troubleshoot this?

**Comment by ant-ds** [comment]
Update: I checked the continuity between the pins and it seems like RXD pin is not connected to the ESP32. I'm not sure if I caused this during soldering, or if this is intentional. I ordered a second board to find out and will update this thread.

**Comment by flaviut** [comment]
Can you please post pictures of the front and back of the entire board, as well as a front and back near the esp32 chip?

**Comment by ant-ds** [comment]
I managed to flash my board by soldering a wire directly to the TXD pin on the ESP32 (the pin marked as "RXD" at the debug pins). 
I should know by Wednesday whether this TXD disconnect is due to a soldering error on my side (FYI, even after I removed my soldered pin, I could not get continuity). I will post pictures of the new rev5 board as soon as it comes in the mail.

**Comment by whoiswes** [comment]
I have a rev 5 board and was able to flash successfully just now.

**Comment by ant-ds** [comment, marked as answer]
I ordered a second board to see if it was my fault that RXD pin was not connected. Just like with my first board, there is also no continuity between the RXD debug pin and the TXD ESP32 pin. I probed a bit further and noticed there is a 102 resistor between the debug pin and the ESP 32. Is this also the case on the other revisions?

I bypassed this resistor by soldering a wire directly to the TXD ESP32 pin. After that, flashing was a breeze.

Pictures of the Rev5 board:
[Front of Rev5 board](https://imgur.com/2k1szsx)
[Back of Rev5 board](https://imgur.com/X5q15Al)
[Resistor on RXD debug pin (to TXD ESP32 Pin)](https://imgur.com/i9nojir)
[My workaround](https://imgur.com/TwAZW0J)

**Reply by flaviut** [reply]
Interesting! I wouldn't expect the 100Ω on that line to have any effect on flashing success. Perhaps your serial adapter has some unusually strong pull-ups?

Anyway, thank you for the pictures, and I'll keep this in mind if anyone else runs into issues.

**Reply by jthoel** [reply]
I was having a problem reading the board/flash/chip and couldn't get it to work.  Connecting up to the ESP line mentioned made it all work great.  Thanks for the pointer!

**Reply by cplagz** [reply]
same issue, CH340G would not read/write, cut back the glue around antenna and held a dupont pin onto TxD ESP32 from RxD TTL, read, backed up and flashed successfully.

**Reply by meetc0** [reply]
Thank you! I noticed I too had no continuity from the Rx pin. My solution was to solder a jumper over the r82 resistor, and use the programming pins as normal. Also, this board uses the 115200 baud rate to connect.

I don't have a dedicated USB to serial, so used an Arduino board instead. Put a resistor between RST and ground to prevent the amtel chip from booting. use 5v and ground as normal, reverse the Rx and Tx pin connections.

**Reply by jklap** [reply]
Just got a new board -- is a Rev 5 and there is zero continuity from the RX header to TX/GPIO1 --tried the same FTDI that I used to flash my previous Vue anyway (not sure what board rev that was). Zero continuity from R82 too. But the chip flashes fast so just held a dupont to GPIO1

**Comment by whoiswes** [comment]
@ant-ds  - FWIW, I was curious and checked - my board is the same configuration/layout as yours (my part num is 501-1-01219_C1).

I did not have any issues flashing (using a CP2101 based USB UART adapter).

Not sure if this is helpful, just adding the data point.

**Comment by ant-ds** [comment]
Interesting, I used a CH340G USB to TTL adapter. I tried both 5V and 3.3V. Maybe it is too weak for this board when the resistor is in place. Good to know!

EDIT my part number is also 501-1-01219_C1. I blurred it out before as I was not sure whether this was unique.

**Comment by Mrkricfalusi** [comment]
I can attest this is also a issue with the Rev 5 board. 

I got my second Vue today and it was a Rev 5 board. Soldered everything up and ran "esptool flash_id" and it identified the chip!

Started backing up the firmware and it stalled at 95%.
restarted. wouldn't connect. booted on the Emporia firmware and worked as expected as stock firmware would.
Tried all the IO1-EN combinations, Rx Tx swap nothing. 
I held a wire to the above mentioned pin on the actually esp32 chip and BINGO everything went smooth, flashed ESPHOME now I got OTA and I'm good. I did not have to solder the wire, just held the pin VERY firm against the TX pin, pulled the firmware off to make sure I had a good connection before flashing ( I don't know how risky that is during flash if it disconnects)

So if you are having issues. give this a try.

**Reply by clowrey** [reply]
I just flashed a brand new (purchased on amazon last week) emporiavue2 rev5 PCB with no issues same as I always have. 

That 6 pin pogo pin adapter + the CH340 serial converter allows you to flash with no pin holding etc as it has the required RTS + CTS pins. 

You do have to cut a surface mount trace and solder the jumper to convert the VCC pin on the serial adapter from 3.3V to 5V. Although it would probably work with 3.3V as the converter appears to have a 3.3V regulator and you would just need a 7 pin pogo adapter then or soldering in a pin header directly to the PCB. 

I don't even bother backing up the original firmware as I know I'm never going back.. 

Also if using the pogo pin adapter - the pads have allot of solder flux residue from the factory that will prevent it from working most likely (it got me for awhile) so you want to clean that up with fresh solder or scrape it off with something carefully.. You can see it still on the 3.3V pad on my image since I don't use that pad. 

![PXL_20221031_202748286](https://user-images.githubusercontent.com/6935928/199104633-21087cb7-4780-4fc2-991e-0c04ef93c146.jpg)
![PXL_20221031_202738486](https://user-images.githubusercontent.com/6935928/199104645-091d9890-1ab4-4f8d-8082-3ae295952cfa.jpg)
![PXL_20221031_202829957](https://user-images.githubusercontent.com/6935928/199105941-10a91322-5c78-47c6-a7cf-42b65599cea0.jpg)
![PXL_20221031_202858870](https://user-images.githubusercontent.com/6935928/199104663-feff55e9-a29b-4529-92dd-42aaca5b603a.jpg)
![PXL_20221031_203423484](https://user-images.githubusercontent.com/6935928/199105505-a01b4acd-9df7-4d97-8ea2-9a7120c0d502.jpg)

**Reply by ozzy** [reply]
Thank you to all for this fantastic work.  It takes me back to os9 and super-sluth disassembler.

Having recently purchased this version I can confirm that I experienced no continuity problems.

I do however have a problem sending mqtt messages.  After ota upload the log shows mqtt connected but no messages are relayed by the broker.

Any ideas anyone ?

best regards
oz

**Comment by khammo01** [comment]
I have a rev5 board and communication starts no problem but I get a corrupt data fatal error at a random interval between 0% and 18% of ROM backup complete. Same problem when directly connecting to the ESP32 Rx pin as described above.

Anyone else have similar issue and suggestion? I am using this cable

https://amzn.to/3qb562A

on a Mac.

**Comment by Cisien** [comment]
I was able to update a rev-5 board without issue using this serial adapter:
https://www.amazon.com/dp/B07BBPX8B8

**Comment by khammo01** [comment]
I got it working. There was some kind of apple silicon bug. Worked fine on Intel Mac. Kurt HammondOn Aug 6, 2023, at 13:18, Chris ***@***.***> wrote:﻿
I was able to update a rev-5 board without issue using this serial adapter:
https://www.amazon.com/dp/B07BBPX8B8

**Comment by wayneconnolly** [comment]
Rev-5 board here also. Got it working easily with the 1O0 method also. See here: https://github.com/emporia-vue-local/esphome/discussions/53#discussioncomment-6894552

**Comment by tzapu** [comment]
i was able to update a rev-5 board using a NodeMCU as a makeshift serial programmer. Read the firmware from a PI and wrote the basic ESPHome image using the web tool from a Mac/Chrome. Shorted IO0 to ESP32 metal shield when I plugged in the USB cable.

**Comment by rederikus** [comment]
I have just successfully flashed my Revision 4 Vue 2 board.  It is working fine.  However along the way I discovered that there is a type 203 (20nF) capacitor in the circuit between the IO0 header and GPIO0 on the ESP32 chip.  This effectively prevents a ground on the header from grounding GPIO0.  Without GPIO0 being grounded the ESP32 will not go into programming mode. The fix was obvioud but a bit fiddly.  I soldered a wire from GPIO0 on the chip to ground (the chip's case) and all was then well.

Here's a couple of pics of what I did.  All other wiring is as described here.
<img width="1000" height="630" alt="s_IHYHxmt--1000" src="https://github.com/user-attachments/assets/67b3ee64-4899-4227-bb75-4a6ea2fef60d" />
<img width="403" height="239" alt="Screenshot 2026-04-21 135319" src="https://github.com/user-attachments/assets/ac52c3c0-f66d-41f5-a3a0-5d311bd02990" />
<img width="2256" height="4000" alt="IMG20260421120101" src="https://github.com/user-attachments/assets/8c68bb15-988c-4321-ba21-6e23503d4a60" />

I suspect that this little modification may well be incorporated into later releases of this product.
